bigml/linear.py

```diff
@@ -0,0 +1,364 @@
+                    # to be used in predictions
+                    self.t_crit = student_t.interval( \
+                        CONFIDENCE,
+                        self.number_of_samples - self.number_of_parameters)[1]
+                    self.xtx_inverse = list( \
+                        np.linalg.inv(np.array(self.xtx_inverse)))
+
+            else:
+                raise Exception("The linear regression isn't finished yet")
+        else:
+            raise Exception("Cannot create the LinearRegression instance."
+                            " Could not find the 'linear_regression' key"
+                            " in the resource:\n\n%s" %
+                            linear_regression)
+
+    def expand_input(self, input_data, unique_terms, compact=False):
+        """ Creates an input array with the values in input_data and
+        unique_terms and the following rules:
+        - fields are ordered as input_fields
+        - numeric fields contain the value or 0 if missing
+        - categorial fields are one-hot encoded and classes are sorted as
+          they appear in the field summary. If missing_count > 0 a last
+          missing element is added set to 1 if the field is missing and 0
+          otherwise
+        - text and items fields are expanded into their elements as found
+          in the corresponding summmary information and their values treated
+          as numerics.
+        """
+        input_array = []
+        for index, field_id in enumerate(self.coeff_ids):
+            field = self.fields[field_id]
+            optype = field["optype"]
+            missing = False
+            new_inputs = []
+            if optype == NUMERIC:
+                if field_id in input_data:
+                    value = input_data.get(field_id, 0)
+                else:
+                    missing = True
+                    value = 0
+                new_inputs = [value]
+            else:
+                terms = getattr(self, EXPANSION_ATTRIBUTES[optype])[field_id]
+                length = len(terms)
+                if field_id in unique_terms:
+                    new_inputs = get_terms_array( \
+                        terms, unique_terms, field, field_id)
+                else:
+                    new_inputs = [0] * length
+                    missing = True
+
+            if field["summary"]["missing_count"] > 0 or \
+                    (optype == CATEGORICAL and \
+                     self.field_codings[field_id].get(DUMMY) is None):
+                new_inputs.append(int(missing))
+
+            if optype == CATEGORICAL:
+                new_inputs = self.categorical_encoding( \
+                    new_inputs, field_id, compact)
+
+            input_array.extend(new_inputs)
+
+        if self.bias or not self.compact:
+            input_array.append(1)
+
+        return input_array
+
+    def categorical_encoding(self, inputs, field_id, compact):
+        """Returns the prediction and the confidence intervals
```

Comment:
is this docstring correct? the name of the method seems weird if so...

Reply:
it isn't... cut and paste strikes again